src/other/index.jl

```diff
@@ -46,14 +46,36 @@ function names!(x::Index, nms::Vector{Symbol}; makeunique::Bool=false)
 end
 
 function rename!(x::Index, nms)
+    xbackup = copy(x)
+    toholder = Dict()
     for (from, to) in nms
         from == to && continue # No change, nothing to do
-        if haskey(x, to)
+        if !haskey(xbackup, from)
+            copy!(x.lookup, xbackup.lookup)
+            x.names .= xbackup.names
+            error("Tried renaming $from to $to, when $from does not exist in the Index.")
```

Comment:
Can you please switch here and below to `throw(ArgumentError(` to raise a more specific exception?